### ISSUE TITLE
Setting `min` and `max` displays calendar events a week earlier than they are scheduled the week of DST

### ISSUE THREAD

**Issue by prestontunnellwilson** [jquense/react-big-calendar#2466]
### Check that this is really a bug

- [X] I confirm

### Reproduction link

https://codesandbox.io/s/react-big-calendar-example-forked-w4u7s2?file=/index.js

### Bug description

Viewing a week of events the week of the daylight savings time shift (in New York timezone) results in events being displayed on hour earlier. This happens when `min` and `max` are specified.

They show the correct time in the Day view, but when November 5th, 2023 (when DST occurred), the events display one hour earlier:

![calendar-event-hour-early-in-week](https://github.com/jquense/react-big-calendar/assets/5190845/906dd8de-c7e0-472b-823f-de06c6076e9f)

You can see that the min and max are different in the day with DST compared to the days without. The code specifies

```jsx
  const min = useMemo(() => new Date(2023, 10, 5, 8, 0, 0, 0), []);
  const max = useMemo(() => new Date(2023, 10, 5, 21, 0, 0, 0), []);
```

But when November 5th, 2023 shows up the min and max are instead 0700 and 2000.

### Expected Behavior

The min's and max's are the same regardless of whether the week includes the Daylight Savings time transition or not, and the calendar events show up at the same time when they are scheduled for according to the Day view.

### Actual Behavior

The min's and max's are shifted an hour earlier the week of DST when in week view and the day of DST when in day view and calendar events for that week in week view show up an hour earlier.

### react-big-calendar version

1.8.5

### React version

18.2.0

### Platform/Target and Browser Versions

macOS Chrome 

### Validations

- [X] Read the [docs](http://jquense.github.io/react-big-calendar/examples).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.
- [X] Make sure this is a react-big-calendar issue and not an implementation issue

### Would you like to open a PR for this bug?

- [ ] I'm willing to open a PR

**Comment by bu6n** [comment]
What localizer are you using? `luxon`? Are you sure this is related to `min` and `max`? If this is possible for you, you could test changing localizers. I fixed this bug for `moment` earlier this year (https://github.com/jquense/react-big-calendar/pull/2374), and it's not present in `date-fns`, but as far as I know still present in `luxon`.

**Comment by prestontunnellwilson** [comment]
> What localizer are you using

This is using `moment`, interestingly enough:

```
    "moment": "2.29.4",
    "moment-timezone": "0.5.38",
```

**Comment by bu6n** [comment]
Could you try with `date-fns` just to see if the problem persists?

**Comment by prestontunnellwilson** [comment]
@bu6n I am sorry for the delay in responding! Trying with `date-fns`, the problem does not persist. I verified this in https://codesandbox.io/s/react-big-calendar-example-forked-g8hk7z?file=/index.js.

That seems to me to point to an issue somewhere in moment or the moment localizer. Does that match what you're thinking?

**Comment by bu6n** [comment]
Yes, I only use `date-fns` now because I had all sort of problems with moment...

**Comment by usmanabdurrehman** [comment]
> [@bu6n](https://github.com/bu6n) I am sorry for the delay in responding! Trying with `date-fns`, the problem does not persist. I verified this in https://codesandbox.io/s/react-big-calendar-example-forked-g8hk7z?file=/index.js.
> 
> That seems to me to point to an issue somewhere in moment or the moment localizer. Does that match what you're thinking?

I am facing the same issue. The issue only occurs when we are using moment.tz.setDefault to set a timezone since the localizer implementation in RBC is using both Date + moment for calculations and even if we set the moment timezone to something like eastern, the Date object would still take the machine timezone which could be anything and that causes issues. 

Your dateFns implementation is working I guess since you arent setting any global timezone using dateFns
